lib/elixir/lib/calendar/time.ex

```diff
@@ -365,12 +365,20 @@ defmodule Time do
   end
 
   @doc """
-  Converts the `Time` struct to a different calendar.
+  Converts given `time` to a different calendar.
 
   Returns `{:ok, time}` if the conversion was successful,
   or `{:error, reason}` if it was not, for some reason.
+
+  ## Examples
+
+  Imagine someone implements `Calendar.Julian`:
+
+      iex> Time.convert(~T[13:30:15], Calendar.Julian)
+      {:ok, %Time{calendar: Calendar.Julian, hour: 13, minute: 30, second: 15, microsecond: {0, 0}}}
+
   """
-  @spec convert(Calendar.time, Calendar.calendar) :: {:ok, Time.t} | {:error, atom}
+  @spec convert(Calendar.time, Calendar.calendar) :: {:ok, t} | {:error, atom}
   def convert(%Time{calendar: calendar} = time, calendar) do
```

Comment:
this is a special clause just for the time in the same calendar as we're converting. There's another more general clause below (not shown in diff by default), and that one does not have `%Time{}`